python_modules/dagster/dagster/_core/definitions/freshness_checks/non_partitioned.py

```diff
@@ -0,0 +1,170 @@
+            severity=severity,
+        )
+    ]
+
+
+def _build_freshness_check_for_asset(
+    asset: Union[CoercibleToAssetKey, AssetsDefinition, SourceAsset],
+    freshness_cron: Optional[str],
+    maximum_lag_minutes: int,
+    severity: AssetCheckSeverity,
+) -> Sequence[AssetChecksDefinition]:
+    checks = []
+    for asset_key in asset_to_keys_iterable(asset):
+
+        @asset_check(
+            asset=asset,
+            description=f"Evaluates freshness for targeted asset. Cron: {freshness_cron}, Maximum lag minutes: {maximum_lag_minutes}.",
+            name="freshness_check",
+        )
+        def the_check(context: AssetExecutionContext) -> AssetCheckResult:
+            check.invariant(
+                context.job_def.asset_layer.asset_graph.get(asset_key).partitions_def is None,
+                "Expected non-partitioned asset",
+            )
+
+            current_time = pendulum.now("UTC")
+            current_timestamp = check.float_param(current_time.timestamp(), "current_time")
+            latest_cron_tick = get_latest_completed_cron_tick(freshness_cron, current_time, None)
+            lower_search_bound = pendulum.instance(latest_cron_tick or current_time).subtract(
+                minutes=maximum_lag_minutes
+            )
+            latest_record = retrieve_latest_record(context.instance, asset_key, partition_key=None)
+            last_updated_timestamp = get_last_updated_timestamp(latest_record)
+
+            if last_updated_timestamp is None:
+                passed = True
+                metadata = {}
+                description = "Could not determine last updated timestamp"
+            else:
+                passed = lower_search_bound.timestamp() <= last_updated_timestamp
+                metadata = {
+                    "dagster/last_updated_timestamp": MetadataValue.timestamp(
+                        last_updated_timestamp
+                    )
+                }
+
+                if passed:
+                    description = None
+                else:
+                    if latest_cron_tick is None:
+                        description = f"Last update was more than {maximum_lag_minutes} minutes ago"
+                    else:
+                        description = (
+                            f"Last update was more than {maximum_lag_minutes} before last cron tick"
+                        )
+                        metadata["dagster/latest_cron_tick"] = MetadataValue.timestamp(
+                            latest_cron_tick
+                        )
+                    metadata["dagster/minutes_late"] = TimestampMetadataValue(
+                        (current_timestamp - last_updated_timestamp) // 60
+                    )
+
+            return AssetCheckResult(
+                passed=passed, severity=severity, description=description, metadata=metadata
+            )
+
+        checks.append(the_check)
+
+    return checks
+
+
+def get_latest_completed_cron_tick(
```

Comment:
This seems generally useful enough that it should go in the same utils file as `reverse_cron_string_iterator`